src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -41,6 +42,8 @@ public class DistributedApplicationBuilder : IDistributedApplicationBuilder
     private const string BuilderConstructingEventName = "DistributedApplicationBuilderConstructing";
     private const string BuilderConstructedEventName = "DistributedApplicationBuilderConstructed";
 
+    private readonly Lazy<Assembly?> _appHostAssembly;
```

Comment:
Do we need a Lazy here? Would it be better to just have a reference to the `DistributedApplicationOptions` instance as a private field, and then below:

```csharp
    /// <inheritdoc />
    public Assembly? AppHostAssembly => _options.Assembly;
```

Reply:
I started out trying to plumb the options that way and landed on this instead for some reason but I can take another look.

Reply:
OK did what you suggested.